tasks.ps1

```diff
@@ -48,23 +48,41 @@ if ($deployYamlsFound.Count -gt 0) {
   Write-Host "mage-runner version: $env:MAGE_RUNNER_VERSION"
 
   nuget sources Add `
-    -Name octopus `
-    -Source https://pkgs.dev.azure.com/USXpress-Inc/CloudOps/_packaging/Octopus/nuget/v3/index.json `
-    -UserName "github-runner" `
-    -Password $env:AZ_DEVOPS_PAT
+  -Name octopus `
+  -Source https://pkgs.dev.azure.com/USXpress-Inc/CloudOps/_packaging/Octopus/nuget/v3/index.json `
+  -UserName "github-runner" `
+  -Password $env:AZ_DEVOPS_PAT
 
   ie nuget install mage-runner `
     -Source octopus `
     -OutputDirectory mage `
     -Version $env:MAGE_RUNNER_VERSION
 
-  Move-Item -Path ./mage/*/mage -Destination ./mage/
-  chmod +x ./mage/mage
+  $S3Bucket = $env:MAGE_S3_BUCKET
+  if ($env:GitVersion_BranchName -eq "master" -or $env:GitVersion_BranchName -eq "main") {
+      $BranchType = "stable"
+  } else {
+      $BranchType = "pre-release"
+  }
+
+  $S3Key = "$BranchType/mage-runner.$env:MAGE_RUNNER_VERSION.zip"
+  $ZipFilePath = "./mage/mage.zip"
+
+  Write-Host "Fetching $S3Key from s3://$S3Bucket/"
+  aws s3 cp "s3://$S3Bucket/$S3Key" $ZipFilePath --force
+
+  if (Test-Path -Path $ZipFilePath) {
```

Comment:
can we add one more condition that checks for exit code for aws s3 command that way we will be sure that no error occured during command.